packages/token/src/tests/erc20/test_erc20.cairo

```diff
@@ -611,3 +650,32 @@ fn assert_state_after_burn(account: ContractAddress, amount: u256) {
     assert_eq!(current_supply, initial_supply - amount);
     assert_eq!(state.balance_of(account), initial_supply - amount);
 }
+
+#[generate_trait]
+impl ERC20HooksSpyHelpersImpl of ERC20HooksSpyHelpers {
+    fn assert_event_before_update(
+        ref self: EventSpy,
+        contract: ContractAddress,
+        from: ContractAddress,
+        recipient: ContractAddress,
+        amount: u256
+    ) {
+        let expected = ERC20MockWithHooks::Event::BeforeUpdate(
+            ERC20MockWithHooks::BeforeUpdate { from, recipient, amount }
+        );
+        self.assert_emitted_single(contract, expected);
+    }
+
+    fn assert_event_after_update(
+        ref self: EventSpy,
+        contract: ContractAddress,
+        from: ContractAddress,
+        recipient: ContractAddress,
+        amount: u256
+    ) {
+        let expected = ERC20MockWithHooks::Event::AfterUpdate(
```

Comment:
What do you think about adding `assert_no_events_left_from(contract)` to these?

Reply:
If you agree, I'd apply this to erc721 as well

Reply:
`update` itself emits a `Transfer` event, so I think this would only make sense on the after hook. But not sure how valuable it is? 
I'll prioritize the other comments + adding the tests to `ERC1155` and then see if I can add this

Reply:
Since we are getting the release today I'll go ahead and merge. Will try to add this to a next PR. Thanks!